shm/Map.cpp

```diff
@@ -0,0 +1,51 @@
+
+#include <iostream>
+
+#include "Map.hpp"
+#include "ShmExceptions.hpp"
```

Comment:
Generally order of includes are: file header, standard library include then other files from project. So it should look like tihs:
#include "Map.hpp"

#include iostream

#include "ShmExceptions.hpp"